x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/RestEsqlTestCase.java

```diff
@@ -71,6 +73,30 @@ public abstract class RestEsqlTestCase extends ESRestTestCase {
 
     private static final List<String> NO_WARNINGS = List.of();
 
+    private static final String MAPPING_ALL_TYPES;
+
+    static {
+        try {
+            InputStream mappingPropertiesStream = RestEsqlTestCase.class.getResourceAsStream("/mapping-all-types.json");
+            String properties = new String(mappingPropertiesStream.readAllBytes(), StandardCharsets.UTF_8);
+            MAPPING_ALL_TYPES = "{\"mappings\": " + properties + "}";
+        } catch (IOException ex) {
+            throw new RuntimeException(ex);
+        }
```

Comment:
You need to close the stream in case of errors - hence use the [try-with-resource](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html) pattern.
Also extract the code into a static method and use that for field initialization:

```java
@BeforeClass
public static void initClass() {
   MAPPING_ALL_TYPES = loadResourceAsString("/mapping-all-types.json");
}

public static String loadResourceAsString(String name) throws Exception {
try (var url = RestEsqlTestCase.class.getResource(name)) {
    return Files.readString(Paths.get(url.toURI()), StandardCharsets.UTF_8));
}
```
There's probably an utility somewhere in ES that does something similar.

Reply:
:facepalm: of course, thank you.

I couldn't find something that _just_ loads a file as text; most utilities seem to directly parse that (e.g. `EsqlTestUtils.loadmapping` and `TypesTests.loadMapping`) or not expose that (e.g. `GeometrySimplifierTests.loadJsonFile`) :/